### DISCUSSION TITLE
Separate API and Client

### DISCUSSION THREAD

**Discussion by jzunigarce** [gbowne1/RadioLogger#46] [Ideas]
I propose to separate the app into two applications. In one it would be the api (currently server folder) and in another we could make a SPA or mount the front with NGINX

**Comment by gbowne1** [comment]
That is a good idea but I think that would take too much work.  I plan on self hosting because I have the bandwidth and the rackspace with Apache2.  Most of our packages are serverside anyhow.   For that to make sense we would have to add more packages to the frontend/client.

**Reply by gbowne1** [reply]
I would be in favor of converting to React instead at some point down the road when we have more completed.

**Comment by jzunigarce** [comment]
What do you think if we move the client folder to server and rename it as views, and move the public folder with the assets(js, css, img) to the same level as views(public) folder. And move move src folder out site src y rename as app or src(and remove the currently src folder)
`/app or src(currently called server, )
 ----all folder and files of server
 ----views(all html files)
/public(currently un client folder, only the css, js, ims)`

**Reply by jzunigarce** [reply]
![image](https://user-images.githubusercontent.com/722001/235330403-9833bbdb-7162-4840-aee3-3c03d2cb9b1f.png)
![image](https://user-images.githubusercontent.com/722001/235330411-70e6d584-0fed-4409-aeca-efd23abc158e.png)

**Reply by gbowne1** [reply]
I don't know that would really help things any.  I am not saying it wouldn't hurt, but I doubt that is really necessary.  If we moved to something like React, by simply installing the react package at some point, that might be a thing.  I think it's fine the way it is.  It will get bundled and built just fine by Babel && Webpack.. in fact the structure we have already is the way its recommended.. all we need to do is add a build folder/directory. 

What actually gets served (by any type of server) is the build.

**Reply by jzunigarce** [reply]
@gbowne1 I Add two different routes: 
- web: localhost:3000 
- api: localhost:3000/api/v1 

For now I only added session verification via token(jwt) to the api routes (currently only the password/reset route is verified token ).

- We need to add in the api the rest of the functionalities for the other Endpoints
- We need to define whether to request a page an authentication mechanism (token or session) will also be handled since those endpoints do not have it or only the verification will be made to the api calls

**Reply by gbowne1** [reply]
Session verification via token will work just fine for now.

How will we consume external APIs from other apps and websites. The scanlog page will need to access to an API.  The hamlog page will need access to an API from the FCC.
